h/api/schemas.py

```diff
@@ -264,17 +192,50 @@ class CreateAnnotationSchema(object):
     """Validate the POSTed data of a create annotation request."""
 
     def __init__(self, request):
-        self.structure = AnnotationSchema(request)
+        self.structure = AnnotationSchema()
+        self.request = request
 
     def validate(self, data):
-        return self.structure.validate(data)
+        appstruct = self.structure.validate(data)
+
+        new_appstruct = {}
+
+        _remove_protected_fields(appstruct)
+
+        new_appstruct['userid'] = self.request.authenticated_userid
+        new_appstruct['target_uri'] = appstruct.pop('uri', u'')
+        new_appstruct['text'] = appstruct.pop('text', u'')
```

Comment:
`AnnotationBasePresenter` will tidy this up on the way out (converting `None` to `''`), but anyone dealing with an `Annotation` model object directly will have to be aware that `annotation.text` might be `None` and not a string.